xmtp/src/client.rs

```diff
@@ -191,9 +192,18 @@ where
 
         self.store.conn().unwrap().transaction(
             |transaction_manager| -> Result<(), ClientError> {
+                self.store.insert_or_ignore_user_with_conn(
```

Comment:
Feels a little wasteful to insert this and then update it a few lines later. Should probably consolidate into a single function that tries to insert and updates the `refresh_timestamp` on conflict